BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/PortalsManager.java

```diff
@@ -21,8 +22,9 @@
 
 	private DatabaseManager db = BetterShardsPlugin.getInstance().getDatabaseManager();
 	private Map<String, Portal> portals;
-	private List<Player> arrivedPlayers = new ArrayList<Player>();
+	private Map<UUID, Long> arrivedPlayers = new TreeMap<UUID,Long>();
```

Comment:
Same issue here; if you need sorting used the externally synchronized SortedMap interfaces or if sort isn't needed use ConcurrentHashMap